tests/test__io_utils.py

```diff
@@ -94,3 +94,91 @@ def _reader(path):
+    pm = PluginManager()
+    plugin = DynamicPlugin("tif-plugin", plugin_manager=pm)
+    path = "something.TIF"
+
+    # reader should be compatible despite lowercase pattern
+    @plugin.contribute.reader(filename_patterns=["*.tif"])
+    def get_read(path):
+        if path.lower() != path:
+            # if this error is raised we can be certain path is unchanged
+            raise ValueError("Given path contains capitals.")
+
+        def read(paths):
+            return [(None,)]
+
+        return read
+
+    with pytest.raises(ValueError, match="Given path contains capitals."):
+        io_utils._read([path], stack=False, _pm=pm)
+
+
+@pytest.mark.parametrize(
+    "path", ["some_zarr_directory.ZARR", "some_zarr_directory.Zarr"]
+)
+def test_read_zarr_variants(path: str):
+    pm = PluginManager()
+    plugin = DynamicPlugin("zarr-plugin", plugin_manager=pm)
+
+    # reader should be compatible despite lowercase pattern
+    @plugin.contribute.reader(filename_patterns=["*.zarr"], accepts_directories=True)
+    def get_read(path):
+        if path.lower() != path:
+            # if this error is raised we can be certain path is unchanged
+            raise ValueError("Given path contains capitals.")
+
+        def read(paths):
+            return [(None,)]
+
+        return read
+
+    with pytest.raises(ValueError, match="Given path contains capitals."):
+        io_utils._read([path], stack=False, _pm=pm)
+
+
+@pytest.mark.parametrize(
+    "path", ["some_two_ext_file.TAR.gz", "some_two_ext_file.TAR.GZ"]
+)
+def test_read_tar_gz_variants(path: str):
+    pm = PluginManager()
+    plugin = DynamicPlugin("tar-gz-plugin", plugin_manager=pm)
+
+    # reader should be compatible despite lowercase pattern
+    @plugin.contribute.reader(filename_patterns=["*.tar.gz"])
+    def get_read(path):
+        if path.lower() != path:
+            # if this error is raised we can be certain path is unchanged
+            raise ValueError("Given path contains capitals.")
+
+        def read(paths):
+            return [(None,)]
+
+        return read
+
+    with pytest.raises(ValueError, match="Given path contains capitals."):
+        io_utils._read([path], stack=False, _pm=pm)
+
+
+@pytest.mark.parametrize("path", ["some_directory.Final", "some_directory.FINAL"])
+def test_read_directory_variants(path: str):
+    pm = PluginManager()
+    plugin = DynamicPlugin("directory-plugin", plugin_manager=pm)
+
+    # reader should be compatible despite lowercase pattern
+    @plugin.contribute.reader(filename_patterns=["*.final"], accepts_directories=True)
+    def get_read(path):
+        if path.lower() != path:
+            # if this error is raised we can be certain path is unchanged
+            raise ValueError("Given path contains capitals.")
+
+        def read(paths):
+            return [(None,)]
+
+        return read
+
+    with pytest.raises(ValueError, match="Given path contains capitals."):
+        io_utils._read([path], stack=False, _pm=pm)
```

Comment:
Currently, this test is similar to Zarr one. My idea when I was requesting it was to check if a plain directory reader would handle a directory with an "extension" 

I'm not sure if it should be  `filename_patterns=["*"]` or lack of this parameter 

```suggestion
@pytest.mark.parametrize("path", ["some_directory.Final", "some_directory.FINAL"])
def test_read_directory_variants_no_ext(path: str):
    pm = PluginManager()
    plugin = DynamicPlugin("directory-plugin", plugin_manager=pm)

    # reader should be compatible despite lowercase pattern
    @plugin.contribute.reader(filename_patterns=["*"], accepts_directories=True)
    def get_read(path):
        if path.lower() != path:
            # if this error is raised we can be certain path is unchanged
            raise ValueError("Given path contains capitals.")

        def read(paths):
            return [(None,)]

        return read

    with pytest.raises(ValueError, match="Given path contains capitals."):
        io_utils._read([path], stack=False, _pm=pm)
```

Reply:
Oh, sorry I didn't get that.
Looking at the code for dirs:
https://github.com/napari/npe2/blob/00d2bc227437a37f276c1ba9fad108a263504469/src/npe2/_plugin_manager.py#L138-L139
So I think in theory the extension shouldn't matter?

But I'm not sure these tests are checking directories effectively? Because `os.path.isdir(path)` will return false in every case.